doc/guides/gcov.rst

```diff
@@ -0,0 +1,53 @@
+.. _gcov:
+
+Test coverage reports via GCOV
+##############################
+
+Overview
+********
+`gcov<https://gcc.gnu.org/onlinedocs/gcc/Gcov.html>`_ is a test coverage program
+used together with the GCC compiler to analyze and create test coverage reports
+for your programs, helping you create more efficient, faster running code and
+discovering untested code paths
+
+In Zephyr, gcov collects coverage profiling data in RAM (and not to a file
+system) while your application is running. Support for gcov collection and
+reporting is limited by available RAM size and so is currently enabled only
+for QEMU emulation of embedded targets.
+
+The purpose of this document is to detail the steps involved in the generation of
+coverage report for embedded targets. The default gcov behavior will continue to
+remain unchanged for targets which use POSIX_ARCH .
+
+Details
+*******
+There are 2 parts to enable this feature. The first is to enable the coverage for the
+device and the second to enable in the test application. As explained earlier the
+code coverage with gcov is a function of RAM available. Therefore ensure that the
+device has enough RAM when enabling the coverage for it. For example a small device
+like frdm_k64f can run a simple test application but the more complex test
+cases which consume more RAM will crash when coverage is enabled.
+
+To enable the device for coverage, select :option:`CONFIG_HAS_COVERAGE_SUPPORT`
+in the Kconfig.board file.
+
+To report the coverage for the particular test application set :option:`CONFIG_COVERAGE`.
+
+Steps to generate code coverage reports
+***************************************
+
+1. Build the code with CONFIG_COVERAGE=y.::
```

Comment:
don't use a period at the end, just ``::``, and add a blank line after this and before the indented text (same on all the other steps below)

Reply:
Thanks for this info.